bpf/capabilitytracer/capability_tracer.c

```diff
@@ -0,0 +1,43 @@
+#include <bpfcore/utils.h>
+#include <gotracer/go_common.h>
+// #include <linux/sched.h>
+
+typedef struct capability_info {
+    int cap;
+    u64 pid;
+    __u8 comm[16];
+    // char comm[TASK_COMM_LEN];
+} capability_info_t;
+
+const capability_info_t *unused_2 __attribute__((unused));
+
+// Temporary tracking of capabilities
+struct {
+    __uint(type, BPF_MAP_TYPE_RINGBUF);
+    __uint(max_entries, 1 << 12);
+} capability_events SEC(".maps");
+
+SEC("kprobe/capable")
+int BPF_KPROBE(beyla_kprobe_capable, int cap) {
+    __u64 pid_tgid = bpf_get_current_pid_tgid();
+    __u32 pid = pid_tgid >> 32;
+
+    capability_info_t *trace = 
+        bpf_ringbuf_reserve(&capability_events, sizeof(capability_info_t), 0);
+    if (trace) {
```

Comment:
```suggestion
    capability_info_t *trace = 
        bpf_ringbuf_reserve(&capability_events, sizeof(capability_info_t), 0);
    if (!trace) {
        return 0;
    }
```

rationale: 
 - [F.56: Avoid unnecessary condition nesting](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#f56-avoid-unnecessary-condition-nesting)
 - https://www.kernel.org/doc/html/next/process/coding-style.html